src/api.py

```diff
@@ -201,25 +232,24 @@ def generate_with_retry(
     ) -> List[UsageReport]:
         """Call the API to generate the next section of text."""
         logging.info(
-            f"Making OpenAI GPT-4 chat completion call on behalf of user with id: {user}"
+            f"Making LiteLLM call on behalf of user with id: {user}"
         )
         options = options or {}
         stopwords = options.get("stop", None)
         functions = options.get("functions", None)
 
-        # record the time before the request is sent
-        start_time = time.time()
+        billing_callback = BillingCallback() if self.steamship_billing else NoopBilling()
 
         @retry(
             reraise=True,
             stop=stop_after_attempt(self.config.max_retries),
             wait=wait_exponential_jitter(jitter=5),
             before_sleep=before_sleep_log(logging.root, logging.INFO),
             retry=(
-                retry_if_exception_type(openai.error.Timeout)
-                | retry_if_exception_type(openai.error.APIError)
-                | retry_if_exception_type(openai.error.APIConnectionError)
-                | retry_if_exception_type(openai.error.RateLimitError)
+                retry_if_exception_type(openai.APITimeoutError)
+                | retry_if_exception(lambda e: isinstance(e, openai.APIError) and not ("does not support parameters" in e.message))
```

Comment:
This is an exception that will never recover, when you supply params to a model that doesn't support them.

You can configure litellm to drop these, but that goes against principle of least surprise.